lib/identity_cache/query_api.rb

```diff
@@ -468,7 +485,23 @@ def prefetch_one_association(association, records)
             prefetch_embedded_association(records, association, details)
             parent_records = records.map(&details[:cached_accessor_name].to_sym).compact
           else
-            raise ArgumentError.new("Non-embedded has_one associations do not support prefetching yet.")
+            ids_to_parent_record = records.each_with_object({}) do |record, hash|
+              child_ids = Array(record.send(details[:cached_id_name]))
+              child_ids.each do |child_id|
```

Comment:
This makes it seem like `record.send(details[:cached_id_name])` might return an array, which I didn't think was true.  I would think that `if child_id = record.send(details[:cached_id_name]` would make more sense than these two lines.